File: LABA4py/text_analyzer.py
import re

class TextAnalyzer:
    def count_words(self, text):
        if text is None:
            raise ValueError("Text cannot be None")
        if not text.strip():
            return 0
        return len(text.strip().split())

    def is_palindrome(self, word):
        if word is None:
            return False
        clean = re.sub(r'\s+', '', word).lower()
        return clean == clean[::-1]

    def count_vowels(self, text):
        if text is None:
            raise ValueError("Text cannot be None")
        vowels = "aeiouyаеёиоуыэюя"
        return sum(1 for char in text.lower() if char in vowels)

    def to_slug(self, text):
        if text is None:
            raise ValueError("Text cannot be None")
        return re.sub(r'\s+', '-', text.lower().strip())

    def check_length(self, text, min_len, max_len):
        if text is None:
            return False
        return min_len <= len(text) <= max_len

File: LABA4java/src/main/java/org/example/TextAnalyzer.java
package org.example;

public class TextAnalyzer {

    public int countWords(String text) {
        if (text == null) throw new IllegalArgumentException("Text cannot be null");
        if (text.trim().isEmpty()) return 0;
        return text.trim().split("\\s+").length;
    }

    public boolean isPalindrome(String word) {
        if (word == null) return false;
        String clean = word.replaceAll("\\s+", "").toLowerCase();
        return clean.equals(new StringBuilder(clean).reverse().toString());
    }

    public int countVowels(String text) {
        if (text == null) throw new IllegalArgumentException("Text cannot be null");
        return (int) text.toLowerCase().chars()
                .filter(c -> "aeiouyаеёиоуыэюя".indexOf(c) != -1)
                .count();
    }

    public String surgeryToSlug(String text) {
        if (text == null) throw new IllegalArgumentException("Text cannot be null");
        return text.toLowerCase().trim().replaceAll("\\s+", "-");
    }

    public boolean checkLength(String text, int min, int max) {
        if (text == null) return false;
        return text.length() >= min && text.length() <= max;
    }
}